core/src/compression/zlib.cc

```diff
@@ -0,0 +1,131 @@
+/*
+** Copyright 2011-2017 Centreon
+**
+** Licensed under the Apache License, Version 2.0 (the "License");
+** you may not use this file except in compliance with the License.
+** You may obtain a copy of the License at
+**
+**     http://www.apache.org/licenses/LICENSE-2.0
+**
+** Unless required by applicable law or agreed to in writing, software
+** distributed under the License is distributed on an "AS IS" BASIS,
+** WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+** See the License for the specific language governing permissions and
+** limitations under the License.
+**
+** For more information : contact@centreon.com
+*/
+
+#include "com/centreon/broker/compression/stream.hh"
+#include "com/centreon/broker/compression/zlib.hh"
+#include "com/centreon/broker/exceptions/corruption.hh"
+#include "com/centreon/broker/logging/logging.hh"
+
+using namespace com::centreon::broker::compression;
+
+/**************************************
+*                                     *
+*      Public Static Methods          *
+*                                     *
+**************************************/
+
+/**
+ * Compression function
+ *
+ * @param data the data to compress.
+ * @param compression_level The compression level, by default -1.
+ *
+ * @return The same data compressed.
+ */
+QByteArray zlib::compress(QByteArray const& data, int compression_level) {
```

Comment:
Please add unit tests.

Reply:
Done for two cases:
* nominal
* when the buffer to compress is empty